zebra/kernel_netlink.c

```diff
@@ -382,6 +385,15 @@ static int netlink_socket(struct nlsock *nl, unsigned long groups,
 	return ret;
 }
 
+/* Make socket for Linux route netlink interface. */
+static int rt_netlink_socket(struct nlsock *nl, unsigned long groups,
```

Comment:
I would just remove this wrapper function and just add the NETLINK_ROUTE to the netlink_socket call.  Too much abstraction for very little gain